src/stores/pacbioLibraries.js

```diff
@@ -0,0 +1,143 @@
+import { defineStore } from 'pinia'
+import useRootStore from '@/stores'
+import { handleResponse } from '@/api/ResponseHelper.js'
+import { groupIncludedByResource, dataToObjectById } from '@/api/JsonApi.js'
+
+/**
+ * 'TODO_LIBRARY_CHANGE': This store needs to be updated to use the new library creation
+ * All the changes, that might be required, are marked with the above comment
+ */
+export const usePacbioLibrariesStore = defineStore('pacbioLibraries', {
+  state: () => ({
+    tagSetChoices: [],
+    tagChoices: {},
+    libraries: [],
+  }),
+
+  getters: {
+    /**Return all libraries */
+    libraryRequest: () => {
+      const rootStore = useRootStore()
```

Comment:
Can you initialise useRootStore() at the top of the file? Instantiated in quite a few places and its not dynamic.

Does libraryRequest need to be a getter? Seems to only be used once or twice in the actions so not sure if its need to be a getter.

Reply:
For the useRootStore comment, I am afraid it is not possible with this way of  defining stores  which  is called as 'option stores' which defines getters, actions and state.
With option store, we need to call the other store in each action or getter where you need it, though that each call to useRootStore  will return the same instance of the store and not creating a new instance each time you call it.
There is another way of defining stores which are 'setup' stores (which is a very similar concept with options api and  composition apis) which allows it. 
I opted for option stores because I thought that will give a more clear transition from how it is defined in VueX store.

Reply:
Ah ok makes sense, yes I have only used the setup/options api store so I was in that mind set when looking at this. Thanks for explaining.